tests_integration/test_integration.py

```diff
@@ -0,0 +1,297 @@
+    new_settings.frac_update_upstream_versions = 1.0
+    new_settings.graph_github_backend_repo = "regro-staging/cf-graph-countyfair"
+    new_settings.conda_forge_org = "conda-forge-bot-staging"
+
+    with use_settings(new_settings):
+        setup_logging(logging.INFO)
+        yield
+
+
+@pytest.fixture
+def disable_container_mode():
+    """
+    Disable container mode for the test.
+    """
+    value_before = os.environ["CF_FEEDSTOCK_OPS_IN_CONTAINER"]
+    os.environ["CF_FEEDSTOCK_OPS_IN_CONTAINER"] = "true"
+    yield
+    os.environ["CF_FEEDSTOCK_OPS_IN_CONTAINER"] = value_before
+
+
+@pytest.fixture(scope="module")
+def repositories_setup():
+    """
+    Set up the repositories for the tests.
+    """
+    setup_repositories.prepare_all_accounts()
+
+
+@pytest.fixture(params=get_all_test_scenario_ids())
+def scenario(request) -> tuple[int, dict[str, TestCase]]:
+    scenario_id: int = request.param
+    close_all_open_pull_requests()
+    reset_cf_graph()
+
+    scenario = get_test_scenario(scenario_id)
+    scenario_pretty_print = {
+        feedstock_name: test_case.__class__.__name__
+        for feedstock_name, test_case in scenario.items()
+    }
+
+    print(f"Preparing test scenario {scenario_id}...")
+    print(f"Scenario: {scenario_pretty_print}")
+
+    run_all_prepare_functions(scenario)
+
+    return scenario_id, scenario
+
+
+def is_proxy_running(port: int, timeout: float = 2.0) -> bool:
+    """
+    Returns if the proxy is running on localhost:port.
+
+    :param port: The port to check.
+    :param timeout: The timeout in seconds.
+
+    :return: A function that returns True if the proxy is running, False otherwise.
+    """
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+        sock.settimeout(timeout)
+        return sock.connect_ex(("localhost", port)) == 0
+
+
+@pytest.fixture
+def mitmproxy(xprocess: XProcess, scenario: tuple[int, dict[str, TestCase]]):
+    scenario_id, _ = scenario
+
+    class MitmproxyStarter(ProcessStarter):
+        args = ["./mock_proxy_start.sh"]
+        timeout = 60
+        popen_kwargs = {"cwd": TESTS_INTEGRATION_DIR}
+        env = os.environ | {
+            "SCENARIO_ID": str(scenario_id),
+            "PYTHONPATH": str(CF_SCRIPTS_ROOT_DIR.resolve()),
+        }
+
+        def startup_check(self):
+            return is_proxy_running(port=8080)
+
+    xprocess.ensure("mitmproxy", MitmproxyStarter)
+
+    yield
+
+    xprocess.getinfo("mitmproxy").terminate()
+
+
+@contextlib.contextmanager
+def in_fresh_cf_graph():
+    """
+    Context manager to execute code within the context with a new clone
+    of cf-graph in a temporary directory.
+    """
+    old_working_dir = os.getcwd()
+
+    with tempfile.TemporaryDirectory() as tmpdir_s:
+        tmpdir = Path(tmpdir_s)
+
+        cf_graph_repo = settings().graph_github_backend_repo
+        subprocess.run(
+            [
+                "git",
+                "clone",
+                "--depth=5",
```

Comment:
why 5?

Reply:
because this is the same setting that prod uses: https://github.com/regro/cf-scripts/blob/d3f591c2c7ac9f09c364c024d019cd7ff4f870d9/autotick-bot/install_bot_code.sh#L46

I'd guess this avoids downloading the entire graph history but still allows going back a few commits.

Reply:
maybe add a reference to this file somewhere here?

Reply:
jup, done in 56e8afe39189e684f3fffd70803338f30902163d